### PR TITLE
fix: use thanos default port in service and containerPort

### PR DESCRIPTION
**Pull request by jan--f**
Before thanos was only listening on localhost. Relying on the default setting allows user to modify that via server side apply and add a proxy chain.

Fixes: https://issues.redhat.com/browse/COO-14

### CODE REVIEW

**Comment by jan--f** [comment]
No reason, just a mistake :)

**Comment by danielmellado** [comment]
hmm it does seem that tests still are targeting the 9090 port ```E0208 11:47:09.149727   35900 portforward.go:409] an error occurred forwarding 9090 -> 9090: error forwarding port 9090 to pod 4d6e4b0f907b07e5f24151733d584f37bb83a6b9bc2483e568a8eaae49b9eb99, uid : failed to execute portforward in network namespace"``` so this would probably have to be changed as well.

**Comment by jan--f** [comment]
Ok at least my local test failure reproduces here too.
```
E0208 11:47:09.149727   35900 portforward.go:409] an error occurred forwarding 9090 -> 9090: error forwarding port 9090 to pod 4d6e4b0f907b07e5f24151733d584f37bb83a6b9bc2483e568a8eaae49b9eb99, uid : failed to execute portforward in network namespace "/var/run/netns/cni-d3ec7578-2ca0-8f70-59dd-2a9a054707e9": failed to connect to localhost:9090 inside namespace "4d6e4b0f907b07e5f24151733d584f37bb83a6b9bc2483e568a8eaae49b9eb99", IPv4: dial tcp4 127.0.0.1:9090: connect: connection refused IPv6 dial tcp6 [::1]:9090: connect: connection refused 
panic: lost connection to pod
```
Not sure what's happening here yet.

**Comment by jan--f** [comment]
> hmm it does seem that tests still are targeting the 9090 port `E0208 11:47:09.149727 35900 portforward.go:409] an error occurred forwarding 9090 -> 9090: error forwarding port 9090 to pod 4d6e4b0f907b07e5f24151733d584f37bb83a6b9bc2483e568a8eaae49b9eb99, uid : failed to execute portforward in network namespace"` so this would probably have to be changed as well.

I don't think so. This error happens in `TestMonitoringStackController/Verify_multi-namespace_support` and the port forward is to prometheus. Here we change the thanos service port.

**Comment by jan--f** [comment]
Or am I just not smart enough to read the output correctly :thinking:

**Comment by jan--f** [comment]
> Or am I just not smart enough to read the output correctly 🤔

Sure looks that way. Testing locally now since here it was failing in a prometheus test.

**Comment by jan--f** [comment]
Ok this is the error I'm seeing locally too:
```
=== RUN   TestMonitoringStackController/Verify_multi-namespace_support
E0208 14:22:19.405068   28162 portforward.go:409] an error occurred forwarding 9090 -> 9090: error forwarding port 9090 to pod b29bce2e1202569aeef6862fdfe16660317d86f50ea82774ee9b1177a236ec14, uid : failed to execute portforward in network namespace "/var/run/netns/cni-a601776d-003d-7bbe-fde4-a5b9daa71b03": failed to connect to localhost:9090 inside namespace "b29bce2e1202569aeef6862fdfe16660317d86f50ea82774ee9b1177a236ec14", IPv4: dial tcp4 127.0.0.1:9090: connect: connection refused IPv6 dial tcp6 [::1]:9090: connect: connection refused 
panic: lost connection to pod
```
Port 9090 is correct in this context and this code didn't change.

**Comment by danielmellado** [comment]
> > hmm it does seem that tests still are targeting the 9090 port `E0208 11:47:09.149727 35900 portforward.go:409] an error occurred forwarding 9090 -> 9090: error forwarding port 9090 to pod 4d6e4b0f907b07e5f24151733d584f37bb83a6b9bc2483e568a8eaae49b9eb99, uid : failed to execute portforward in network namespace"` so this would probably have to be changed as well.
> 
> I don't think so. This error happens in `TestMonitoringStackController/Verify_multi-namespace_support` and the port forward is to prometheus. Here we change the thanos service port.

Yep, I was a bit too quick into this as the port was just matching the one you changed over there, but after checking the tests code this *should* be unrelated.

**Comment by danielmellado** [comment]
@sthaha mind taking a look at this? thanks!

**Comment by jan--f** [comment]
I have disabled the test for evicting alertmanager pods under the PDB for now. I'll keep investigating this, but I also wonder whether this is a test we need to run. I'd argue we test k8s api here and that is out of scope for this operator.

File: test/run-e2e.sh [hunk 1]
```diff
@@ -154,7 +155,7 @@ run_e2e() {
 	watch_obo_errors "$obo_error_log" &
 
 	local ret=0
-	go test -v -failfast -timeout $TEST_TIMEOUT ./test/e2e/... --retain=true | tee "$LOGS_DIR/e2e.log" || ret=1
+	go test -v -failfast -timeout $TEST_TIMEOUT ./test/e2e/... -run "$RUN_REGEX" -count 1 | tee "$LOGS_DIR/e2e.log" || ret=1
```
Comment (danielmellado) [inline test/run-e2e.sh:158]: neat on the addition on the regex!

**Pull request merged**